demo/pkg/subgraphs/countries/subgraph/resolver.go

```diff
@@ -11,6 +11,6 @@ import (
 // It serves as dependency injection for your app, add any dependencies you require here.
 
 type Resolver struct {
-	NC  *nats.Conn
-	mux sync.Mutex
+	NatsConnectionBySourceName map[string]*nats.Conn
```

Comment:
How about passing a map[string]PubSub?
That'll be more flexible in the future to add other PubSubs, instead of having to add Kafka, MQTT, etc...